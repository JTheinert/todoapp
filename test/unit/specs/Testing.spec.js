import Testing from '../../../src/components/testing.vue'
import { mount } from '@vue/test-utils'

describe('Testing.vue', () => {
  let wrapper = {}

  beforeEach(() => {
    wrapper = mount(Testing)
  })

  test('button click should increment the count', () => {
    expect(wrapper.vm.count).toBe(0)
    const button = wrapper.find('button')
    button.trigger('click')
    expect(wrapper.vm.count).toBe(1)
  })

  test('function has been called', () => {
    let wrapper = mount(Testing, {
      propsData: {
        count: 0
      }
    })
    const increment = jest.fn()

    wrapper.setMethods({
      increment: increment
    })
    wrapper.find('button').trigger('click')

    expect(increment).toHaveBeenCalled()
  })
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
