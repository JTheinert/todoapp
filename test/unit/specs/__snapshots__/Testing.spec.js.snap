// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing.vue renders correctly 1`] = `
<div>
  <span
    class="count"
  >
    0
  </span>
   
  <button>
    Increment
  </button>
</div>
`;
